src/sentry/api/bases/organization_events.py

```diff
@@ -69,6 +69,7 @@ def get_snuba_query_args_v2(self, request, organization, params):
         fields = request.GET.getlist('field')[:]
         aggregations = []
         groupby = request.GET.getlist('groupby')
+        having = request.GET.getlist('having')
```

Comment:
Is request.GET the query string of the HTTP get without being validated? 
If yes, I would discourage from picking a query parameter from the request and building the snuba query without sanitizing it (in this case allowing only specific fields to end in the "having/groupby" clause.
Building a query from non sanitized input may be a security vulnerability (SQL injection in the worst case). In this specific case I can imagine someone trying to cause a DOS by fabricating very expensive snuba query the product is not supposed to run.

Reply:
@getsentry/app-backend  thoughts? My thoughts here is that it'll always be limited by the project  and datetime. It's also readonly. There is a timeout on our end that gives up if the query doesn't return in under a certain time.

Reply:
To be clear, I would not push back on this specific PR (picking the groupby fields from the GET request was already in the code before).
I think this is a practice we should avoid in general. Whitelisting the queries we want the client to be able to run (instead of blacklisting what we do not want the client to run) is very easy to achieve and maintain. It also makes the system as a whole robust to changes that can happen outside this piece of code (like someone introducing a way around the project id/ time filter). 
Otherwise, for any change that impacts the way we query snuba/clickhouse we would have to re-evaluate whether the change is allowing the client to run dangerous queries.

Reply:
I've also had concerns about us passing bare strings to Clickhouse, since I don't know whether we have a driver at some point sanitizing input.

I would say that I think it's the responsibility of the Snuba service to provide sanitation/protection from sql injection attacks. Although I do agree that we should define aggregations and other potential queries on our end instead of letting users just perform any query that they want.